src/lib/transition_router/transition_router.ml

```diff
@@ -134,7 +134,38 @@ module Make (Inputs : Inputs_intf) = struct
        it didn't receive any transition from network for a while.
        Then it went to the second epoch and it could propose at
        the second epoch. *)
-    if Consensus.Hooks.is_genesis @@ Coda_base.Block_time.now time_controller
+    if
+      try
+        Consensus.Hooks.is_genesis @@ Coda_base.Block_time.now time_controller
```

Comment:
it's a bit hacky to catch the exception I think, can we make `is_genesis` return something like a `(bool, [> Time_to_soon]) Result.t` or make `is_genesis` a tri-state:

```ocaml
`After
`During
`Before
```

Reply:
The exception is raised where the epoch number is calculated. There are a bunch of calls to that point.

We can catch the exception further down, like in `is_genesis`, and return a value such as you suggest. OK?

Reply:
hmm, yeah this is fine then. Can you put a comment about why you're catching the exception though?